src/rewriters/Rewriter.h

```diff
@@ -24,17 +24,19 @@ class Rewriter {
         auto size = Idx(logic.getPterm(root).getId()) + 1;
         std::vector<char> done;
         done.resize(size, 0);
-        std::unordered_map<PTRef, PTRef, PTRefHash> substitutions;
+        Map<PTRef, PTRef, PTRefHash> substitutions;
```

Comment:
Can this be `Logic::SubstMap`? Or even better `SubstitutionConfig::SubMap`?

Reply:
Do you dislike the template parameters here or do you think we'd have to go through its keys?  I don't see the need for that now.  We could also revert it back to `std::unordered_map` as far as I understand.

Reply:
No, quite the opposite. It's just that `Logic::SubstMap` is exactly `Map<PTRef, PTRef, PTRefHash>` so I was wondering if we should use that consistently.

Reply:
Ah right!  `Logic::SubstMap` is `MapWithKeys<PTRef,PTRef,PTRefHash>` so right now I would say it's better not to mix them.

Reply:
OK, I agree!